### ISSUE TITLE
kpromo timed out when recursively signing

### ISSUE THREAD

**Issue by puerco** [kubernetes-sigs/promo-tools#892]
#### What happened:

During the 1.28.0-alpha.1 release, we had another breakage when `kpromo` was signing images recursively, it timed out talking to the registry:

```
time="23:18:19.455" level=fatal msg="run `cip run`: promote images: signing images: signing image europe-west9-docker.pkg.dev/k8s-artifacts-prod/images/conformance@sha256:be2dc70c66df0e6149ef4fd37d4029f6097d195eb6e90693c6d9a1cbc26ea8f5: sign reference: europe-west9-docker.pkg.dev/k8s-artifacts-prod/images/conformance@sha256:be2dc70c66df0e6149ef4fd37d4029f6097d195eb6e90693c6d9a1cbc26ea8f5: recursively signing: signing digest: Get \"https://europe-west9-docker.pkg.dev/v2/\": context deadline exceeded (and 5 more errors)" diff=815ms
```

**Comment by ameukam** [comment]
FWIW, this is a first release after release tooling now uses cosign v2.

/milestone v1.28
cc @cpanato

**Comment by puerco** [comment]
Not on the promoter, this is going on in https://github.com/kubernetes-sigs/promo-tools/pull/889

**Comment by saschagrunert** [comment]
It's the recursive signing option which we added support for in https://github.com/kubernetes-sigs/promo-tools/pull/868, I'll double check what we can do about the timtouts.

**Comment by saschagrunert** [comment]
Maybe it would be enough to increase the timeout of 3 minutes to something higher?

https://github.com/kubernetes-sigs/promo-tools/blob/20f8804b3ec0747a2516f92e0668a36fd74b3733/internal/promoter/image/sign.go#L108


@puerco do you have some logs which may indicate what the overall timeout value is?

https://github.com/kubernetes-sigs/release-sdk/blob/878a1fb69d00ac61c1799ee8d95c020fbdf218b4/sign/options.go#L97

https://github.com/kubernetes-sigs/release-sdk/blob/878a1fb69d00ac61c1799ee8d95c020fbdf218b4/sign/options.go#L109C19-L113

**Comment by xmudrii** [comment]
@saschagrunert No, that's all we have. You can see the logs here: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-k8sio-image-promo/1663678115915763712
